src/test_utils.rs

```diff
@@ -186,6 +187,37 @@ impl TestedDialects {
         statements
     }
 
+    /// The same as [`statements_parse_to`] but it will strip semicolons from the SQL text.
+    pub fn statements_without_semicolons_parse_to(
+        &self,
+        sql: &str,
+        canonical: &str,
+    ) -> Vec<Statement> {
+        let sql_without_semicolons = sql
+            .replace("; ", " ")
+            .replace(" ;", " ")
+            .replace(";\n", "\n")
+            .replace("\n;", "\n")
+            .replace(";", " ");
```

Comment:
wouldn't this last statement (replace `";"` with `" "` handle all the above? Or is the idea that newlines are important ?

Can you please add comments explaining the rationale for these substitutions?

Reply:
I'm preserving the newline/whitespace around the statement delimiter because I'm trying to avoid being opinionated on whether they're meaningful or not _in this helper_. i.e., the helper should only strip semicolons and should avoid other transformations. I can add a comment

Reply:
Good point. At some point I was probably testing this in conjunction with https://github.com/apache/datafusion-sqlparser-rs/pull/1809 and might have gotten it from there. Regardless, I'll simplify it here per your suggestion.